src/Microsoft.AspNetCore.SystemWebAdapters/Adapters/ProxyHeaderModule.Framework.cs

```diff
@@ -34,12 +34,53 @@ public void Init(HttpApplication context)
 
             context.BeginRequest += (s, e) => UseOptions(((HttpApplication)s).Context.Request);
         }
+
+        context.PreSendRequestHeaders += (s, e) => FixupLocationHeader(((HttpApplication)s).Context);
+    }
+
+    private static void FixupLocationHeader(HttpContext context)
+    {
+        // Some scenarios (such as OIDC auth) will store a URL (including the local host/port rather than the forwarded one)
+        // and later redirect users to that URL. If a response contains a Location header, ensure that the correct scheme,
+        // host, and port are used.
+        if (context.Response.Headers["Location"] is { } location)
```

Comment:
why include this fix up here rather than in the remote app module?

Reply:
I could see it living either place. If OIDC code worked the way I wanted it to, this would have been handled by `UseHeaders` in this module (by setting SERVER_NAME, etc.). Because that doesn't work, we have the 'backup plan' of checking Location headers here.

We could put this in the remote app authentication module since auth scenarios are the ones we know of now that lead to incorrect Location response headers, but if there were ever any other buggy modules/handlers/etc. that resulted in Location headers pointing at the old app we'd want to fix those up, too.

Reply:
See above. I believe this is fixed in #92

Reply:
Awesome. I'll see if I can revert this part of the PR then.

Reply:
I've confirmed that #92 fixes the issue with OIDC not respecting forwarded host/port, so I've reverted this part of the PR.